packages/shorebird_cli/lib/src/commands/patch/linux_patcher.dart

```diff
@@ -0,0 +1,167 @@
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/archive/archive.dart';
+import 'package:shorebird_cli/src/artifact_builder.dart';
+import 'package:shorebird_cli/src/artifact_manager.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/code_signer.dart';
+import 'package:shorebird_cli/src/commands/commands.dart';
+import 'package:shorebird_cli/src/common_arguments.dart';
+import 'package:shorebird_cli/src/extensions/arg_results.dart';
+import 'package:shorebird_cli/src/logging/logging.dart';
+import 'package:shorebird_cli/src/patch_diff_checker.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/release_type.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/src/base/process.dart';
+import 'package:shorebird_code_push_protocol/shorebird_code_push_protocol.dart';
+
+/// {@template linux_patcher}
+/// Functions to create a linux patch.
+/// {@endtemplate}
+class LinuxPatcher extends Patcher {
+  /// {@macro linux_patcher}
+  LinuxPatcher({
+    required super.argParser,
+    required super.argResults,
+    required super.flavor,
+    required super.target,
+  });
+
+  @override
+  Future<void> assertPreconditions() async {}
+
+  @override
+  Future<DiffStatus> assertUnpatchableDiffs({
+    required ReleaseArtifact releaseArtifact,
+    required File releaseArchive,
+    required File patchArchive,
+  }) async {
+    // TODO
+    return const DiffStatus(
+      hasAssetChanges: false,
+      hasNativeChanges: false,
+    );
+    // return patchDiffChecker.confirmUnpatchableDiffsIfNecessary(
+    //   localArchive: patchArchive,
+    //   releaseArchive: releaseArchive,
+    //   archiveDiffer: const WindowsArchiveDiffer(),
+    //   allowAssetChanges: allowAssetDiffs,
+    //   allowNativeChanges: allowNativeDiffs,
+    // );
+  }
+
+  @override
+  Future<File> buildPatchArtifact({String? releaseVersion}) async {
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+
+    final buildAppBundleProgress = logger.detailProgress(
+      'Building Linux app with Flutter $flutterVersionString',
+    );
+
+    final releaseDir = Directory(
+      p.join(
+        shorebirdEnv.getFlutterProjectRoot()!.path,
+        'build',
```

Comment:
I wonder if we end up with some sort of projectPath('foo', 'bar', 'baz') type helpers for these.